src/folderlistmodel.cpp

```diff
@@ -364,6 +429,23 @@ FolderListModel::FolderStandardType FolderListModel::folderTypeFromId(const QMai
     return NormalFolder;
 }
 
+bool FolderListModel::isStandardFolder(const QMailFolderId &id) const
+{
+    FolderStandardType folderType = folderTypeFromId(id);
+    return folderType == InboxFolder || folderType == DraftsFolder
+            || folderType == SentFolder || folderType == TrashFolder || folderType == OutboxFolder;
+}
+
+FolderListModel::FolderItem* FolderListModel::createFolderItem(const QModelIndex &idx, const QMailFolderId &mailFolderId,
+                                                               FolderStandardType mailFolderType,
+                                                               const QMailMessageKey &folderMessageKey)
+{
+    FolderItem *item = new FolderItem(idx, mailFolderId, mailFolderType, folderMessageKey, 0);
+    item->index = createIndex(m_folderList.count(), 0, item);
```

Comment:
To make this easier to follow, it would be better if you appended the new folder to 'm_folderList' in this function as well as accessing its count, or, do both those things in the other function and pass the row value into this function.